tableauserverclient/server/endpoint/workbooks_endpoint.py

```diff
@@ -199,7 +199,8 @@ def _get_wb_preview_image(self, workbook_item):
 
     # Publishes workbook. Chunking method if file over 64MB
     @api(version="2.0")
-    def publish(self, workbook_item, file_path, mode, connection_credentials=None):
+    @parameter_added_in(as_job='3.0')
+    def publish(self, workbook_item, file_path, mode, as_job=False, connection_credentials=None):
```

Comment:
For backwards compatibility, you should put `as_job` as the last argument because if someone has written the code as `publish(workbook_item, file_path, 'append', creds)`, this would break them.  You need to do the same thing for datasource.